src/sinks/mod.rs

```diff
@@ -87,6 +87,8 @@ pub mod redis;
 pub mod s3_common;
 #[cfg(feature = "sinks-sematext")]
 pub mod sematext;
+#[cfg(feature = "sinks-sftp")]
```

Comment:
We should also gate this behind `unix` as the sftp opendal integration states that it only works for unix.

```suggestion
#[cfg(all(unix, feature = "sinks-sftp"))]
```